dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaClustersTask.java

```diff
@@ -0,0 +1,124 @@
+  private static final Logger LOG = LoggerFactory.getLogger(ClouderaClustersTask.class);
+
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  public ClouderaClustersTask() {
+    super("clusters.json");
+  }
+
+  @CheckForNull
+  @Override
+  protected Void doRun(
+      TaskRunContext context, @Nonnull ByteSink sink, @Nonnull ClouderaManagerHandle handle)
+      throws Exception {
+    CloseableHttpClient httpClient = handle.getHttpClient();
+
+    ApiClusterListDto clusterList;
+
+    if (context.getArguments().getCluster() != null) {
+      final String clusterName = context.getArguments().getCluster();
+      try (CloseableHttpResponse clusterResponse =
+          httpClient.execute(new HttpGet(handle.getApiURI() + "/clusters/" + clusterName))) {
+
+        ApiClusterDto cluster =
+            objectMapper.readValue(
+                EntityUtils.toString(clusterResponse.getEntity()), ApiClusterDto.class);
+
+        clusterList = new ApiClusterListDto();
+        clusterList.setItems(ImmutableList.of(cluster));
+      }
+    } else {
+      LOG.info("'--cluster' argument wasn't provided. Collect all available clusters.");
+
+      try (CloseableHttpResponse clustersResponse =
+          httpClient.execute(new HttpGet(handle.getApiURI() + "/clusters"))) {
+        String clustersJson = EntityUtils.toString(clustersResponse.getEntity());
+
+        clusterList = objectMapper.readValue(clustersJson, ApiClusterListDto.class);
+      }
+    }
+
+    try (Writer writer = sink.asCharSink(StandardCharsets.UTF_8).openBufferedStream()) {
+      writer.write(objectMapper.writeValueAsString(clusterList));
+    }
+
+    List<ClouderaClusterDTO> clusters = new ArrayList<>();
+    for (ApiClusterDto item : clusterList.getItems()) {
+      String clusterId = requestClusterIdByName(httpClient, handle.getBaseURI(), item.getName());
+      clusters.add(ClouderaClusterDTO.create(clusterId, item.getName()));
+    }
+    LOG.info(
+        "Dump metadata for clusters: {}",
+        clusters.stream().map(ClouderaClusterDTO::getName).collect(Collectors.toList()));
+
+    handle.initClusters(clusters);
+    return null;
+  }
+
+  private String requestClusterIdByName(
+      CloseableHttpClient httpClient, URI baseUri, String clusterName) throws Exception {
+    String requestUrl = baseUri + "/cmf/clusters/" + clusterName + "/status.json";
+
+    try (CloseableHttpResponse clusterStatus = httpClient.execute(new HttpGet(requestUrl))) {
+      String responseBody = EntityUtils.toString(clusterStatus.getEntity());
+      if (HttpStatus.SC_OK != clusterStatus.getStatusLine().getStatusCode()) {
+        LOG.warn(
+            "Can't receive cluster [{}] status by url [{}]. Response status is [{}] and body: {}",
+            clusterName,
+            requestUrl,
+            clusterStatus.getStatusLine().getStatusCode(),
+            responseBody);
+
+        return null;
+      } else {
+        JsonNode jsonNode = objectMapper.readTree(responseBody);
```

Comment:
Should we have parsed this from the entity's InputStream, and only used EntityUtils.toString() in the error-path?

Reply:
Well, good point.  It might be combined with mockserver in tests and should looks good. 
I added it to the scope http://b/377651080

Reply:
done